api/rosetta/data_transaction_integration_test.go

```diff
@@ -0,0 +1,555 @@
+			wantRosettaErrorDetails:     map[string]interface{}{"index": uint64(testHeight), "hash": invalidBlockHash},
+		},
+		{
+			name: "unknown block",
+			request: rosetta.TransactionRequest{
+				NetworkID: defaultNetwork(),
+				BlockID: identifier.Block{
+					Index: lastHeight + 1,
+				},
+				TransactionID: testTx,
+			},
+
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorUnknownBlock,
+			wantRosettaErrorDescription: fmt.Sprintf("block index is above last indexed block (last: %d)", lastHeight),
+			wantRosettaErrorDetails:     map[string]interface{}{"index": uint64(lastHeight + 1), "hash": ""},
+		},
+		{
+			name: "mismatched block height and hash",
+			request: rosetta.TransactionRequest{
+				NetworkID: defaultNetwork(),
+				BlockID: identifier.Block{
+					Index: 44,
+					Hash:  testBlockHash,
+				},
+				TransactionID: testTx,
+			},
+
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidBlock,
+			wantRosettaErrorDescription: fmt.Sprintf("block hash does not match known hash for height (known: %s)", knownHeaders(44).ID().String()),
+			wantRosettaErrorDetails:     map[string]interface{}{"index": uint64(44), "hash": testBlockHash},
+		},
+		{
+			name: "missing transaction id",
+			request: rosetta.TransactionRequest{
+				NetworkID: defaultNetwork(),
+				BlockID:   testBlock,
+				TransactionID: identifier.Transaction{
+					Hash: "",
+				},
+			},
+
+			wantStatusCode:              http.StatusBadRequest,
+			wantRosettaError:            configuration.ErrorInvalidFormat,
+			wantRosettaErrorDescription: "transaction identifier: hash field is empty",
+		},
+		{
+			name: "missing transaction id",
+			request: rosetta.TransactionRequest{
+				NetworkID: defaultNetwork(),
+				BlockID:   testBlock,
+				TransactionID: identifier.Transaction{
+					Hash: trimmedTxHash,
+				},
+			},
+
+			wantStatusCode:              http.StatusBadRequest,
+			wantRosettaError:            configuration.ErrorInvalidFormat,
+			wantRosettaErrorDescription: fmt.Sprintf("transaction identifier: hash field has wrong length (have: %d, want: %d)", len(trimmedTxHash), validBlockHashLen),
+		},
+		{
+			name: "invalid transaction id",
+			request: rosetta.TransactionRequest{
+				NetworkID: defaultNetwork(),
+				BlockID:   testBlock,
+				TransactionID: identifier.Transaction{
+					Hash: invalidTxHash,
+				},
+			},
+
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidTransaction,
+			wantRosettaErrorDescription: "transaction hash is not a valid hex-encoded string",
+			wantRosettaErrorDetails:     map[string]interface{}{"hash": invalidTxHash},
+		},
+		// TODO: add - transaction that has no events/transfers
+		// TODO: add - transaction that does not exist in a block
+		// 	=> https://github.com/optakt/flow-dps/issues/195
```

Comment:
:+1:

Reply:
I left the TODOs for the valid test cases that should be added once this is implemented, remove the one that only works for the current behaviour